### DISCUSSION TITLE
[OP Stack Deployment] Inquiry on Withdrawal Fee Calculation, Charging Responsibility, and Optional Nature of Withdrawals

### DISCUSSION THREAD

**Discussion by DavidRomm** [ethereum-optimism/developers#105] [Deploying the OP Stack]
### Issue Description

Regarding the transfer of assets from the Optimism network to the Ethereum mainnet,
I seek clarity on several key points relating to the fee structure. 

Firstly, what formula is used to calculate the withdrawal fees, 
and are the determining factors similar to those affecting regular transaction fees, 
such as gas prices and network demand? 

Secondly, who charges these withdrawal fees – is it the responsibility of the end user 
or does the sequencer handle these costs?
 
Additionally, I am interested in understanding whether the withdrawal process to the Ethereum mainnet is optional. 
Can users opt to keep their assets on the L2 Optimism network indefinitely, 
or if there are certain circumstances where they would be required or advised
to move their assets to the Ethereum mainnet (L1)?

Insight into how this payment process is managed within the Optimism network would be greatly appreciated.

**Comment by imtipi** [comment]
> and are the determining factors similar to those affecting regular transaction fees,
> such as gas prices and network demand?

gas fee only


> Can users opt to keep their assets on the L2 Optimism network indefinitely,
> or if there are certain circumstances where they would be required or advised
> to move their assets to the Ethereum mainnet (L1)?

sequencer has no power to force you withdraw to Ethereum mainnet.


for more information about withdrawal:https://docs.optimism.io/stack/protocol/withdrawal-flow

**Reply by DavidRomm** [reply]
Thank you for the previous response and the link to the
withdrawal information. I'd like to explore further the specifics of
the L1 data fee during the withdrawal process from
Optimism to Ethereum mainnet. 

I would appreciate clarification on whether the end user is
responsible for these L1 data fees during withdrawal, 
or if there are any circumstances where
these costs are handled differently?

Understanding the nuances of the L1 data fee in the 
withdrawal process would greatly enhance our 
understanding of the costs involved for 
users moving assets from L2 to L1.

**Reply by imtipi** [reply]
> responsible for these L1 data fees during withdrawal,

initial withdraw:you pay the gas to send a transaction to L2
prove withdraw:you pay the gas to send a transaction to L1 ethereum
only gas

**Reply by sbvegan** [reply]
Following up here, the user is responsible for paying for the whole withdrawal. Information on the withdrawal process is outlined in this [blog post](https://blog.oplabs.co/two-step-withdrawals/). 

The user makes three transactions:

1. L2 withdrawal tx
2. L1 prove tx
3. L1 finalize tx

**Comment by sbvegan** [comment]
Hey @DavidRomm, I'm going to close this discussion for now. Please feel free to mark your answer and we can reopen the discussion if you have more questions.
